src/cursor.coffee

```diff
@@ -304,13 +304,14 @@ class Cursor extends Model
     else
       {row, column} = @getScreenPosition()
 
+      originalRow = row
       while columnCount > column and row > 0
         columnCount -= column
         column = @editor.lineTextForScreenRow(--row).length
         columnCount-- # subtract 1 for the row move
 
       column = column - columnCount
-      @setScreenPosition({row, column})
+      @setScreenPosition({row, column}, wrapAtSoftNewlines: originalRow == row)
```

Comment:
My intuition would be that we *always* wrap at soft newlines when moving right, never when moving left. If we're trying to move rightward, landing in a phantom token moves us further rightward. If we're trying to move leftward, the opposite is true.